### PR TITLE
feat: Release notes for Codacy Self-hosted 14.0.0 DOCS-674

### PR DESCRIPTION
**Pull request by Ngwerume**
Adds auto-generated release notes for Codacy Self-hosted 14.0.0

### :eyes: Live preview
https://feat-release-notes-self-hosted-14-0--docs-codacy.netlify.app/release-notes/self-hosted/self-hosted-v14.0.0/

### :construction: To do
- [x] Update the variable `extra.codacy_self_hosted_version` in `mkdocs.yml` to the new Codacy Self-hosted version
- [x] Add the new release notes page to `mkdocs.yml` and `docs/release-notes/index.md`
- Not required ~~Add the breaking changes section if the release is a new major version~~
- [x] Review generated release notes
- [x] Review list of issues with missing release notes
- [x] Ensure that links point to documentation for new Codacy Self-hosted version
- [x] Review links to changelogs of updated tools
- [x] Update the CLI tool versions and update the variable `extra.codacy_coverage_reporter_version` in `mkdocs.yml`
- [x] Update release date and remove `TODO` comments

### CODE REVIEW

**Review by heliocodacy — APPROVED** [review]

File: docs/release-notes/self-hosted/self-hosted-v14.0.0.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+Follow the steps below to upgrade to Codacy Self-hosted v14.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v14.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:
+
+    -   [Codacy Analysis CLI 7.9.5](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.9.5)
+    -   [Codacy Coverage Reporter 13.10.15](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.10.15)
```
Comment (claudiacarpinteiro) [inline docs/release-notes/self-hosted/self-hosted-v14.0.0.md:26]: @Ngwerume I know you already validated and you said "the newer version do not work, due to the nature of updates we are doing on SH which do not update the API."... 🙏 

But I still don't understand this downgrade: on the previous [Self-hosted v13.0.0](https://docs.codacy.com/release-notes/self-hosted/self-hosted-v13.0.0/) we used [Codacy Coverage Reporter 13.13.14](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.13.14).
Shouldn't we keep this same version? 🤔

Comment (nicklem) [reply docs/release-notes/self-hosted/self-hosted-v14.0.0.md:26]: We could add a note. Something like: `Codacy is downgrading to Codacy Coverage Reporter version 13.10.15. Versions above 13.10.15 are currently incompatible with recent updates.`

@Ngwerume @claudiacarpinteiro Thoughts?

Comment (Ngwerume) [reply docs/release-notes/self-hosted/self-hosted-v14.0.0.md:26]: @claudiacarpinteiro we ran into issues with 13.13.14 and resolved with the affected customers, the open end here was we did not clarify with you to change the documents accordingly. 
@nicklem We can change the wording to something like Codacy Coverage Reporter for Self Hosted works with version 13.10.15 but I am ok with how you want to present this.
All SH from v12 should be using the same coverage reporter as we did not upgrade the APIs only changes to tools and k8s bumps.  As we did not update the API, the never version of coverage reporter fails as the API is not available

**Review by claudiacarpinteiro — APPROVED** [review]
LGTM! 👍

**Pull request merged**